conformance/google-java/src/test/kotlin/com/connectrpc/conformance/Conformance.kt

```diff
@@ -49,36 +49,38 @@ import org.junit.runner.RunWith
 import org.junit.runners.Parameterized
 import org.junit.runners.Parameterized.Parameters
 import org.testcontainers.containers.GenericContainer
+import org.testcontainers.containers.wait.strategy.HostPortWaitStrategy
 import java.time.Duration
 import java.util.Base64
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
-
 @RunWith(Parameterized::class)
 class Conformance(
-    private val protocol: NetworkProtocol
+    private val clientProtocol: NetworkProtocol,
+    private val serverProtocol: NetworkProtocol
 ) {
-    private lateinit var connectClient: ProtocolClient
+    private lateinit var protocolClient: ProtocolClient
     private lateinit var shortTimeoutConnectClient: ProtocolClient
     private lateinit var unimplementedServiceClient: UnimplementedServiceClient
-    private lateinit var testServiceConnectClient: TestServiceClient
+    private lateinit var testServiceClient: TestServiceClient
 
     companion object {
-        const val CONFORMANCE_VERSION = "0b07f579cb61ad89de24524d62f804a2b03b1acf"
+        private const val CONFORMANCE_VERSION = "0b07f579cb61ad89de24524d62f804a2b03b1acf"
 
         @JvmStatic
-        @Parameters(name = "protocol")
-        fun data(): Iterable<NetworkProtocol> {
+        @Parameters(name = "client={0},server={1}")
+        fun data(): Iterable<Array<NetworkProtocol>> {
             return arrayListOf(
-                NetworkProtocol.CONNECT,
-                NetworkProtocol.GRPC
+                arrayOf(NetworkProtocol.CONNECT, NetworkProtocol.CONNECT),
+                arrayOf(NetworkProtocol.GRPC, NetworkProtocol.CONNECT),
+                arrayOf(NetworkProtocol.GRPC, NetworkProtocol.GRPC)
```

Comment:
Now these are the combination of client protocol + server protocol we test. The specific combination which was untested before was a gRPC server.